rpc_util.go

```diff
@@ -914,3 +923,91 @@ const (
 )
 
 const grpcUA = "grpc-go/" + Version
+
+const (
+	level0PoolMaxSize = 16
+	level1PoolMaxSize = level0PoolMaxSize * 16
+	level2PoolMaxSize = level1PoolMaxSize * 16 //   4 KB
+	level3PoolMaxSize = level2PoolMaxSize * 16 //  64 KB
+	level4PoolMaxSize = level3PoolMaxSize * 16 //   1 MB
```

Comment:
Given that this is going to a feature which only advanced users will end up using, would it make sense to accept the buffer sizes from the user (as part of the dial option) instead of hard-coding it?

If we end up going with such an approach, the pool of pools (named `bytesPool`) would need to be initialized with a dynamic set of pools, instead of static ones like here.

@hueypark @dfawley : thoughts?

Reply:
I have implemented a `sharedRecvBufferPool` that users can create, so this is no longer an issue for discussion.